ls1intum/Artemis
# Further improvements to the student course overview

### Checklist

### Motivation and Context

We didn't display a graph if there were only exercises with bonus points. If there was nothing to display there was just an empty space.

### Description

We now display a text if there is no graph to display, and a graph if there are bonus points (even when there are no "normally" reachable points otherwise). Known limitations: You can't use the canvas like a link now (clicking it directly works however). The animations for the graph are skipped now too, to reduce the amount of visual "noise".

Also updated the course statistics page to have a red instead of grey color for missing points:
![grafik](https://user-images.githubusercontent.com/72132281/115165464-4a864800-a0ae-11eb-8370-262c775be582.png)

### Steps for Testing

1. Look at the students overview of courses
2. Check that bonus points are included in the statistics
3. Check that a text is displayed if there is no score to be reached and no bonus points

### Screenshots

Without (bonus) points in the course:
![grafik](https://user-images.githubusercontent.com/72132281/115165366-e19ed000-a0ad-11eb-80c2-c4586a059a2a.png)

With only bonus points in the course:
![grafik](https://user-images.githubusercontent.com/72132281/115165373-e794b100-a0ad-11eb-9824-01d5ee7e2938.png)
